cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +38,54 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
-	if !cmdUtil.IsValidProfile(profileName) {
+	if cmdUtil.IsValidProfile(profileName) {
+		//need to check if the current active profile has a VM
+		if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+			err := unsetCurrentOcContext()
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+		}
+
 		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
-	} else {
-		err := cmdUtil.SetOcContext(profileName)
+		fmt.Printf("Profile '%s' set as active profile\n", profileName)
+
+		err = cmdUtil.SetOcContext(profileName)
 		if err != nil {
 			if glog.V(2) {
 				fmt.Println(fmt.Sprintf("%s", err.Error()))
 			}
-			fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName, unableToSetOcContextErrorMessage))
+			fmt.Println(fmt.Sprintf("No oc CLI context is activated for profile, because the '%s' VM does not exist or is not running. Run 'minishift start' first.", profileName))
+		}
+	} else {
+		if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
+			err := unsetCurrentOcContext()
+			if err != nil {
+				fmt.Println(err.Error())
+			}
```

Comment:
The piece of code is repeated both in if and else. Which is wrong. It should be present before the if/else starts. 
```
if cmdUtil.DoesVMExist(config.AllInstancesConfig.ActiveProfile) {
			err := unsetCurrentOcContext()
			if err != nil {
				fmt.Println(err.Error())
			}
		}
```